diff --git a/node_modules/buffer-equal-constant-time/index.js b/node_modules/buffer-equal-constant-time/index.js
index 5462c1f..94977b8 100644
--- a/node_modules/buffer-equal-constant-time/index.js
+++ b/node_modules/buffer-equal-constant-time/index.js
@@ -2,6 +2,7 @@
 'use strict';
 var Buffer = require('buffer').Buffer; // browserify
 var SlowBuffer = require('buffer').SlowBuffer;
+var hasSlowBuffer = SlowBuffer && SlowBuffer.prototype;
 
 module.exports = bufferEq;
 
@@ -28,14 +29,22 @@ function bufferEq(a, b) {
 }
 
 bufferEq.install = function() {
-  Buffer.prototype.equal = SlowBuffer.prototype.equal = function equal(that) {
+  Buffer.prototype.equal = function equal(that) {
     return bufferEq(this, that);
   };
+
+  if (hasSlowBuffer) {
+    SlowBuffer.prototype.equal = function equal(that) {
+      return bufferEq(this, that);
+    };
+  }
 };
 
 var origBufEqual = Buffer.prototype.equal;
-var origSlowBufEqual = SlowBuffer.prototype.equal;
+var origSlowBufEqual = hasSlowBuffer ? SlowBuffer.prototype.equal : undefined;
 bufferEq.restore = function() {
   Buffer.prototype.equal = origBufEqual;
-  SlowBuffer.prototype.equal = origSlowBufEqual;
+  if (hasSlowBuffer) {
+    SlowBuffer.prototype.equal = origSlowBufEqual;
+  }
 };
